pkg/ingester/ingester.go

```diff
@@ -393,6 +398,11 @@ func New(cfg Config, limits *validation.Overrides, ingestersRing ring.ReadRing,
 	i.metrics = newIngesterMetrics(registerer, cfg.ActiveSeriesMetrics.Enabled, i.getInstanceLimits, i.ingestionRate, &i.inflightPushRequests, &i.inflightPushRequestsBytes)
 	i.activeGroups = activeGroupsCleanupService
 
+	if cfg.CircuitBreakerConfig.Enabled {
+		// We create an inactive circuit breaker, which will be activated on a successful completion of starting.
+		i.circuitBreaker = newCircuitBreaker(cfg.CircuitBreakerConfig, false, logger, registerer)
+	}
```

Comment:
Given you're passing the entire `cfg.CircuitBreakerConfig` to `newCircuitBreaker`, and `nil` instance of `*circuitBreaker` is valid, can we move this logic to `newCircuitBreaker`?

Otherwise _this_ method is taking decisions based on _that_ entity's config and if it's false, it relies on _that_ entity's nil behaviour.

Reply:
I have moved the logic in `newCircuitBreaker()`.